microstream/src/main/java/io/micronaut/microstream/conf/EmbeddedStorageManagerFactory.java

```diff
@@ -69,7 +69,9 @@ public EmbeddedStorageManager createEmbeddedStorageManager(EmbeddedStorageFounda
                 throw new DisabledBeanException("Please, define a bean of type " + EmbeddedStorageConfigurationProvider.class.getSimpleName() + " by name qualifier: " + name);
             }
             EmbeddedStorageConfigurationProvider configuration = beanContext.getBean(EmbeddedStorageConfigurationProvider.class, Qualifiers.byName(name));
-            storageManager.setRoot(InstantiationUtils.instantiate(configuration.getRootClass()));
+            if (configuration.getRootClass() != null) {
+                storageManager.setRoot(InstantiationUtils.instantiate(configuration.getRootClass()));
+            }
```

Comment:
@sdelamo I had to add this, as storage being used as a cache has an `one.microstream.collections.types.XTable` as the root object

And we cannot create one of these using `InstantiationUtils` as none of it's implementations have constructors

This breaks the NonNull contract with the root-class configuration

Reply:
Already done in this PR https://github.com/micronaut-projects/micronaut-microstream/pull/23/files#diff-ffe605a8297d0845f69eca25a47b0c79eb2d96e0571e83294ff3ce570c2b2256R37